### PR TITLE
Add {BasicOutput, Output}::is_implicit_account

### PR DESCRIPTION
**Pull request by thibault-martinez**
https://github.com/iotaledger/tips/blob/tip42/tips/TIP-0042/tip-0042.md#creation-semantic-transaction-validation-rules

### CODE REVIEW

**Review by Thoralf-M — DISMISSED** [review]

**Review by kwek20 — DISMISSED** [review]

File: sdk/src/types/block/output/basic.rs [hunk 1]
```diff
@@ -320,6 +320,14 @@ impl BasicOutput {
 
         None
     }
+
+    /// Checks whether the basic output is an implicit account.
+    pub fn is_implicit_account(&self) -> bool {
+        self.unlock_conditions.len() == 1
```
Comment (Alex6323) [inline sdk/src/types/block/output/basic.rs]: any unlock condition does suffice?

Comment (thibault-martinez) [reply sdk/src/types/block/output/basic.rs]: You can't build a BasicOutput without an address so if there is only one UC, it has to be an address

File: sdk/src/types/block/output/mod.rs [hunk 2]
```diff
@@ -252,6 +252,15 @@ impl Output {
         }
     }
 
+    /// Checks whether the output is an implicit account.
+    pub fn is_implicit_account(&self) -> bool {
+        if let Output::Basic(output) = self {
+            output.is_implicit_account()
+        } else {
+            false
+        }
```
Comment (Alex6323) [inline sdk/src/types/block/output/mod.rs:261]: could be made into a one-liner I think:
```rust
matches!(self, Output::Basic(output) if output.is_implicit_account())
```

Comment (thibault-martinez) [reply sdk/src/types/block/output/mod.rs:261]: Are you strongly opinionated about that? I think the if/let version is easier to humanly parse

File: sdk/src/types/block/output/basic.rs (hunk 1 above)
Comment (Alex6323) [reply sdk/src/types/block/output/basic.rs]: sorry, wasn't clear enough, the question was whether we should be explicit here rather then just checking that there is a UC (which then has to be an address UC)

File: sdk/src/types/block/output/mod.rs (hunk 2 above)
Comment (Alex6323) [reply sdk/src/types/block/output/mod.rs:261]: no, not at all, just wanted to mention it if you'ld prefer it a bit shorter. I try to remember your preference.

**Review by Alex6323 — DISMISSED** [review]

**Review by kwek20 — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**